tests/protocol/test_hostplugin.py

```diff
@@ -543,6 +545,86 @@ def test_put_status_unhealthy_signal_permanent(self, patch_http_put, patch_http_
         self.assertFalse(obj['Observations'][0]['IsHealthy'])
         self.assertEqual('GuestAgentPluginStatus', obj['Observations'][0]['ObservationName'])
 
+    @patch("azurelinuxagent.common.protocol.hostplugin.HostPluginProtocol.should_report", return_value=True)
+    @patch("azurelinuxagent.common.protocol.healthservice.HealthService.report_host_plugin_extension_artifact")
+    def test_report_fetch_health(self, patch_report_artifact, patch_should_report):
+        host_plugin = self._init_host()
+        host_plugin.report_fetch_health(uri='', is_healthy=True)
+        self.assertEqual(0, patch_should_report.call_count)
+
+        host_plugin.report_fetch_health(uri='http://169.254.169.254/extensionArtifact', is_healthy=True)
+        self.assertEqual(0, patch_should_report.call_count)
+
+        host_plugin.report_fetch_health(uri='http://168.63.129.16:32526/status', is_healthy=True)
+        self.assertEqual(0, patch_should_report.call_count)
+
+        self.assertEqual(None, host_plugin.fetch_last_timestamp)
+        host_plugin.report_fetch_health(uri='http://168.63.129.16:32526/extensionArtifact', is_healthy=True)
+        self.assertNotEqual(None, host_plugin.fetch_last_timestamp)
+        self.assertEqual(1, patch_should_report.call_count)
+        self.assertEqual(1, patch_report_artifact.call_count)
+
+    @patch("azurelinuxagent.common.protocol.hostplugin.HostPluginProtocol.should_report", return_value=True)
+    @patch("azurelinuxagent.common.protocol.healthservice.HealthService.report_host_plugin_status")
+    def test_report_status_health(self, patch_report_status, patch_should_report):
+        host_plugin = self._init_host()
+        self.assertEqual(None, host_plugin.status_last_timestamp)
+        host_plugin.report_status_health(is_healthy=True)
+        self.assertNotEqual(None, host_plugin.status_last_timestamp)
+        self.assertEqual(1, patch_should_report.call_count)
+        self.assertEqual(1, patch_report_status.call_count)
+
+    def test_should_report(self):
```

Comment:
This is test clarifies a lot of things for me - nice!

Reply:
Awesome.